### PR TITLE
Move Settings load to StartupActions from PostStartupActions #1840

### LINKED ISSUE #1834: Organelle tooltip not translatable
I have translated the organelle descriptions, but the tooltip is still english, except for the reaction name (yellow)
grafik

### LINKED ISSUE #1840: Move Settings load to StartupActions from PostStartupActions
to make sure language option is applied earlier. If applying some setting doesn't work that early it should be moved to be applied with an invoke similarly to how I already did the graphics options setting on initial load.

### PR DESCRIPTION
**Pull request by Sourciluss667**
Closes #1840 
fixes #1834

### CODE REVIEW

**Comment by Sourciluss667** [comment]
Settings load but don't apply, I have this error : 
`void Settings.ApplyAll(): System.NullReferenceException: Object reference not set to an instance of an object.`
I see the error come from all apply function and i don't understand how to resolve this :(

**Comment by hhyyrylainen** [comment]
It's probably because the applies set properties on them. You need to isolate the parts that use an autoload part and use that Invoke approach the same as used for graphics options to apply them. While doing that you also need to ensure that language option is absolutely applied immediately.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
See my comment: https://github.com/Revolutionary-Games/Thrive/pull/1843#issuecomment-728149205

**Comment by Sourciluss667** [comment]
Hello hello !
Invoke approach on `ApplyGraphicsSettings` crash the game, and don't define how to debug that with godot :/
I try to change order of apply functions and I see when `*.Instance` is called, the game crash.
So I think `Invoke.Instance` crash too, and I don't now how to resolve that :/
If you can help me a little thx :)

**Comment by hhyyrylainen** [comment]
Invoke is an autoload as well. It should be moved to be the first autoload.

**Comment by Sourciluss667** [comment]
Okay now I understand how autoload work with Godot ! Invoke and SimulationParameters autoload before StartupActions, so language settings apply :) Do you want a condition in StartupActions to check if Language is apply ?

**Comment by hhyyrylainen** [comment]
> Do you want a condition in StartupActions to check if Language is apply ?

I'm not sure what you mean exactly?

**Comment by Sourciluss667** [comment]
"While doing that you also need to ensure that language option is absolutely applied immediately"
I think you want a if else to verify if language is apply correctly, but i think if ApplyLanguageSettings() don't apply the settings, the game crash, so it's useful ?

(Again sorry for my bad english, I'm trying to do my best to be understandable)

**Comment by hhyyrylainen** [comment]
It's enough to make sure that the language applying code is ran. In fact I don't think the godot language server set language method even returns an error if it fails. So there isn't even a way to check that the language was applied correctly.

File: project.godot [hunk 1]
```diff
@@ -30,9 +30,9 @@ config/windows_native_icon="res://assets/misc/icon.ico"
 
 [autoload]
 
-StartupActions="*res://src/engine/StartupActions.cs"
-SimulationParameters="*res://simulation_parameters/SimulationParameters.cs"
 Invoke="*res://src/engine/Invoke.cs"
+SimulationParameters="*res://simulation_parameters/SimulationParameters.cs"
+StartupActions="*res://src/engine/StartupActions.cs"
```
Comment (hhyyrylainen) [inline project.godot:35]: Unless there's a reason to change the order of these two. They should be kept in the same relative order, so StartupActions should be before simulation parameters like it was before.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: project.godot (hunk 1 above)
Comment (Sourciluss667) [reply project.godot:35]: At the end of ApplyLanguageSettings() in Settings.cs, we have SimulationParameters.Instance.ApplyTranslations() require SimulationParameters to load before or the game crash. If I understand, ApplyLanguageSettings need to be called directly and not with Invoke, so I don't have the choice to move that ?

Comment (hhyyrylainen) [reply project.godot:35]: That's probably correct. It's fine to leave as is.
I initially decided to put startup actions before simulation parameters in order to allow something like a mod loader to modify simulation parameter loading options.

**Comment by hhyyrylainen** [comment]
This just needs some playtesting to make sure this works correctly.

File: src/engine/StartupActions.cs [hunk 2]
```diff
@@ -19,5 +19,9 @@ private StartupActions()
         // Print the logs folder to see in the output where they are stored
         GD.Print("Game logs are written to: ", PathUtils.Join(userDir, Constants.LOGS_FOLDER_NAME),
             " latest log is 'log.txt'");
+
+        // Load settings here, to make sure locales etc. are applied to the main loaded scene
```
Comment (Kasterisk) [inline src/engine/StartupActions.cs:23]:
```suggestion
        // Load settings here, to make sure locales etc. are applied to the main loaded and autoloaded scenes
```

**Review by Kasterisk — DISMISSED** [review]

**Comment by Kasterisk** [comment]
Seems to work fine and looks like #1834 is fixed as well

**Review by hhyyrylainen — APPROVED** [review]
Code looks fine.

**Comment by hhyyrylainen** [comment]
I think the issues @Kasterisk had with this were also solved?

**Comment by hhyyrylainen** [comment]
I'll go ahead and merge this.

**Pull request merged**